scheduler/server/priority_based_scheduling_alg.go

```diff
@@ -0,0 +1,240 @@
+package server
+
+import (
+	"fmt"
+	"math"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/twitter/scoot/common/stats"
+)
+
+type PriorityBasedAlg struct {
+	PriorityRatios           []int // array of priority ratios where array index == priority
+	taskModCounter           float64
+	jobsByPriority           [][]*jobState // list of jobs at each priority
+	runRatios                []float64     // when to run each priority
+	roundRobinJobIndex       []int         // for each priority, track the index of the job whose task should be assigned next
+	numBarredFromLowPriority int           // number of nodes reserved for non-lowest priority tasks
+}
+
+func MakePriorityBasedAlg(ratios []int) *PriorityBasedAlg {
+	pbs := &PriorityBasedAlg{
+		PriorityRatios: ratios,
+		taskModCounter: 1,
+	}
+
+	// initialize the runRatios array
+	// runRatio[pN] represents that we want to run tasks for pN 1 out of runRatio[pN] tasks
+	// when taskModCounter == runRatio[pN] a task from a job with that priority will be scheduled
+	pbs.runRatios = make([]float64, len(pbs.PriorityRatios))
+	pbs.runRatios[0] = 1
+	pbs.roundRobinJobIndex = make([]int, len(pbs.PriorityRatios))
+	for i := 1; i < len(pbs.PriorityRatios); i++ {
```

Comment:
Drew, you were right, multiplying the ratios is not needed.... we can just use the values directly as long as they start with 1 and there are no duplicates.